File: 백준/Silver/1920. 수 찾기/수 찾기.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;


public class Main {

    private static int find(int x, int n, int[] arr){
        int st =0;
        int en =n-1;

        while(st<=en){
            int mid= (st+en)/2;
            if(arr[mid] <x){
                st = mid +1;
            }else if(arr[mid] >x){
                en = mid-1;
            }else return 1;
        }
        return 0;
    }


    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(br.readLine());
        int[] arr = new int[n];

        StringTokenizer st = new StringTokenizer(br.readLine());
        
        for(int i=0; i< n; i++){
            arr[i] = Integer.parseInt(st.nextToken());
        }
        
        Arrays.sort(arr);
        
        int m = Integer.parseInt(br.readLine());

        //정답ㅇ르 StringBuilder for return 
        
        StringBuilder sb = new StringBuilder();

        st = new StringTokenizer(br.readLine());
        for(int i=0; i<m; i++){
            
            int x = Integer.parseInt(st.nextToken());

            sb.append(find(x,n, arr)).append("\n");
        }
        System.out.println(sb);
    }
}

File: 백준/Silver/1920. 수 찾기/수 찾기.cc
#include <iostream>
#include <algorithm>
#include <queue>
#include <set>
#include <map>

using namespace std;
int arr[100000];
int N;
int M;

int find(int x) {
    int st = 0;
    int en =N-1;

    while(st<=en) {//st가 커지는 순간 종료
        int mid = (st+en)/2;
        if(arr[mid]<x) {
            st = mid+1;
        }else if(arr[mid]>x) {
            en = mid-1;
        }else return 1;

    }

    return 0;
}


int main() {
    ios_base::sync_with_stdio(0);
    cin.tie(0);

    cin >>N;
    for(int i=0; i< N; i++) {
        cin >> arr[i];
    }

    sort(arr, arr+N);

    cin >> M;
    for(int i=0; i< M; i++) {
        int x ; cin>>x;

        cout << find(x)<< "\n";
    }

}
